dlt/extract/hints.py

```diff
@@ -84,13 +88,32 @@ class TResourceHints(TResourceHintsBase, total=False):
 
 
 class HintsMeta:
-    __slots__ = ("hints", "create_table_variant")
+    __slots__ = ("hints", "create_table_variant", "data_item_format")
 
-    def __init__(self, hints: TResourceHints, create_table_variant: bool) -> None:
+    def __init__(
+        self,
+        hints: TResourceHints,
+        create_table_variant: bool,
+    ) -> None:
         self.hints = hints
         self.create_table_variant = create_table_variant
 
 
+class SqlModel(NamedTuple):
+    query: str
+    dialect: Optional[str] = None
+
+    @classmethod
+    def from_sqlglot(cls, query: str, dialect: Optional[str] = None) -> "SqlModel":
```

Comment:
The correct name would be `from_query_string`, no? This is independent of sqlglot.